proposals/collection-literals.md

```diff
@@ -112,23 +112,46 @@ Collection literals are [target-typed](https://github.com/dotnet/csharplang/blob
-* [*Span types*](#span-types)
-* Types with a suitable [`Construct` method](#construct-methods)
-* Types that implement `System.Collections.IDictionary` - *dictionary [collection initializers](#collection-initializers)*
-* Interface types _`I<TKey, TValue>`_ implemented by `System.Collections.Generic.Dictionary<TKey, TValue>` (e.g. `IDictionary<TKey, TValue>`, `IReadOnlyDictionary<TKey, TValue>`)
-* Types that implement `System.Collections.IEnumerable` - [*collection initializers*](#collection-initializers)
-* Interface types _`I<T>`_ implemented by `System.Collections.Generic.List<T>` (e.g. `IEnumerable<T>`, `IList<T>`, `IReadOnlyList<T>`)
+* To a *type* that implements `System.Collections.IDictionary` where:
+  * The *type* contains an applicable instance constructor that can be invoked with no arguments or invoked with a single argument for the 0-th parameter where the parameter has type `System.Int32` and name `capacity`.
+  * For each element `Ei` there is an applicable instance indexer for the key value pair from:
+    * If `Ei` is an *expression element* with type `dynamic` or `System.Collections.Generic.KeyValuePair<Ki, Vi>`, the values of applicable properties `Key` and `Value` from `Ei`.
+    * If `Ei` is a *dictionary element* `Ki:Vi`, the values `Ki` and `Vi`.
+    * If `Ei` is a *spread element* with *iteration type* `dynamic` or `System.Collections.Generic.KeyValuePair<Ki, Vi>`, the values of applicable properties `Key` and `Value` from the *iteration variable*.
+
+* To an *interface type* *`I<K, V>`* where `I<TKey, TValue>` is implemented by `System.Collections.Generic.Dictionary<TKey, TValue>` and where for each element `Ei`:
+  * If `Ei` is an *expression element* then the type of `Ei` is `dynamic`, or the type of `Ei` is a type `System.Collections.Generic.KeyValuePair<Ki, Vi>` and there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
+  * If `Ei` is a *dictionary element* `Ki:Vi` there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
+  * If `Ei` is a *spread element* with *iteration type* `dynamic`, or `System.Collections.Generic.KeyValuePair<Ki, Vi>` where there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
+
+* To a *type* that implements `System.Collections.IEnumerable` where:
+  * The *type* contains an applicable instance constructor that can be invoked with no arguments or invoked with a single argument for the 0-th parameter where the parameter has type `System.Int32` and name `capacity`.
+  * No element `Ei` is a *dictionary element*.
+  * For each element `Ei` there is an applicable instance or extension method `Add` for the single argument from:
+    * If `Ei` is an *expression element*, the expression `Ei`.
+    * If `Ei` is a *spread element*, the *iteration variable* of the spread element.
+
+* To an *interface type* *`I<T0>`* where `I<T>` is implemented by `System.Collections.Generic.List<T>` and where for each element `Ei`:
```

Comment:
`T0` is normally a nullability thing, why not `I<T>`?

Reply:
This is a specific substitution for `T` rather than the type parameter `T`.